File: LC_56.py
# 56. Merge Intervals

class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        if intervals == None or len(intervals) == 0:
            return []
        intervals.sort()
        result = [intervals[0]]
        
        for i in range(len(intervals)):
            if result[-1][1] >= intervals[i][0]:
                result[-1][1] = max(result[-1][1], intervals[i][1])
            else:
                result.append(intervals[i])
    
        return result

File: LC_56.java
// 56. Merge Intervals

class Solution {
    public int[][] merge(int[][] intervals) {
        if (intervals.length <= 1)
            return intervals;
        Arrays.sort(intervals, (arr1, arr2) -> Integer.compare(arr1[0], arr2[0]));
        
        List<int[]> result = new ArrayList<>();
        result.add(intervals[0]);
        
        for (int i = 0; i < intervals.length; i++) {
            if (result.get(result.size() - 1)[1] >= intervals[i][0])
                result.get(result.size() - 1)[1] = Math.max(intervals[i][1], result.get(result.size() - 1)[1]);
            else
                result.add(intervals[i]);
            
        }
        
        return result.toArray(new int[result.size()][]);
    }
}
